### PR TITLE
fixed the completions of wpscan

### PR DESCRIPTION
**Pull request by tbhaxor**
## Description
 Fixed the completions for wpscan tool

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: completion/available/wpscan.completion.bash [hunk 1]
```diff
@@ -1,15 +1,62 @@
 #!/usr/bin/bash
 
 if command -v wpscan > /dev/null; then
```
Comment (NoahGorny) [inline completion/available/wpscan.completion.bash:3]: can you convert this to `_command exists`? if its a binary, you can also use `_binary_exists`

**Comment by NoahGorny** [comment]
nice work @tbhaxor !
as I always say, feel free to open up a following cleanup PR :smile:

File: completion/available/wpscan.completion.bash (hunk 1 above)
Comment (tbhaxor) [reply completion/available/wpscan.completion.bash]: Done

**Comment by tbhaxor** [comment]
> nice work @tbhaxor !
> as I always say, feel free to open up a following cleanup PR

Should i checkout from this branch or fresh master branch @NoahGorny ?

**Comment by NoahGorny** [comment]
I guess you should open from this branch.
Thank you for doing this!

File: completion/available/wpscan.completion.bash [hunk 2]
```diff
@@ -1,15 +1,62 @@
...
-        COMPREPLY=()
-        for _opt_ in ${OPTS[@]}; do
-            if [[ "$_opt_" == "$2"* ]]; then
-                COMPREPLY+=("$_opt_")
-            fi
-        done
+if _command_exists wpscan > /dev/null; then
+
+    _wpscan_completions()   {
```
Comment (cornfeedhobo) [inline completion/available/wpscan.completion.bash:5]: Is there a reason this name was changed? If not, could you preserve the original function name `__wpscan_completion`? It's the convention most bash completions use and I'd like to stick with it.

@NoahGorny Maybe this is something we should add to the docs?

**Review by cornfeedhobo — COMMENTED** [review]
Thanks for the great work. I've left one comment.

File: completion/available/wpscan.completion.bash (hunk 2 above)
Comment (NoahGorny) [reply completion/available/wpscan.completion.bash:5]: From `contributing.rst`:
```rst
* Internal functions that aren't to be used by the end user should start with an underscore, e.g. ``_my-new-internal-function``.
```

Maybe we should improve this section

**Review by NoahGorny — CHANGES_REQUESTED** [review]
The completion does not seem to work after your PR, it seems pretty broken. @tbhaxor can you test locally and see if it runs smoothly on your end?

File: completion/available/wpscan.completion.bash (hunk 2 above)
Comment (tbhaxor) [reply completion/available/wpscan.completion.bash:5]: @cornfeedhobo Done

File: completion/available/wpscan.completion.bash [hunk 3]
```diff
@@ -1,15 +1,62 @@
...
-    __wpscan_completion()  {
-        local OPTS=("--help --hh --version --url --ignore-main-redirect --verbose --output --format --detection-mode --scope --headers --user-agent --vhost --random-user-agent --user-agents-list --http-auth --max-threads --throttle --request-timeout --connect-timeout --disable-tlc-checks --proxy --proxy-auth --cookie-string --cookie-jar --cache-ttl --clear-cache --server --cache-dir --update --no-update --wp-content-dir --wp-plugins-dir --wp-version-detection --main-theme-detection --enumerate --exclude-content-based --plugins-list --plugins-detection --plugins-version-all --plugins-version-detection --themes-list --themes-detection --themes-version-all --themes-version-detection --timthumbs-list --timthumbs-detection --config-backups-list --config-backups-detection --db-exports-list --db-exports-detection --medias-detection --users-list --users-detection --passwords --usernames --multicall-max-passwords --password-attack --stealthy")
-        COMPREPLY=()
-        for _opt_ in ${OPTS[@]}; do
-            if [[ "$_opt_" == "$2"* ]]; then
-                COMPREPLY+=("$_opt_")
-            fi
-        done
+if _command_exists wpscan > /dev/null; then
```
[the remark this answers is not in the document]
Comment (tbhaxor) [reply completion/available/wpscan.completion.bash:3]: hi @NoahGorny there is some problem with compgen. On my local dev env I am getting `bash: compgen: -t: invalid option name` when ever I am pressing TAB twice.

Could you help me out here?

[the remark this answers is not in the document]
Comment (NoahGorny) [reply completion/available/wpscan.completion.bash:3]: I think you need to have another go at this fix, as it is fairly broken, and I am not sure why.

[the remark this answers is not in the document]
Comment (tbhaxor) [reply completion/available/wpscan.completion.bash:3]: @NoahGorny I figured it out. I know and I am feeling so embarrassed. It's a trivial bug actually :stuck_out_tongue:

[the remark this answers is not in the document]
Comment (tbhaxor) [reply completion/available/wpscan.completion.bash]: Done

File: completion/available/wpscan.completion.bash [hunk 4]
```diff
@@ -1,15 +1,62 @@
...
+
```
[the remark this answers is not in the document]
Comment (tbhaxor) [reply completion/available/wpscan.completion.bash]: Done

**Comment by tbhaxor** [comment]
@NoahGorny All done. Please review the PR

**Comment by NoahGorny** [comment]
Sadly, it does not work yet :cry: 
You can try and fix it again @tbhaxor 

simplescreenrecorder-2021-01-27_23 56 09

**Comment by tbhaxor** [comment]
> Sadly, it does not work yet 
> You can try and fix it again @tbhaxor
> 
> simplescreenrecorder-2021-01-27_23 56 09

oops, will check this weekend then :P

**Pull request closed without merging**